tools/clang_tidy/lib/clang_tidy.dart

```diff
@@ -171,23 +189,82 @@ class ClangTidy {
     return repo.changedFiles;
   }
 
+  Iterable<T> _takeShard<T>(Iterable<T> values, int id, int shardCount) sync*{
+    int count = 0;
+    for(final T val in values) {
+      if (count % shardCount == id) {
+        yield val;
+      }
+      count++;
+    }
+  }
+
+  Iterable<_SetStatusCommand> _calcIntersection(Iterable<Command> items, Iterable<List<Command>> sets) sync* {
+    bool allSetsContain(Command command) {
+      for (final List<Command> set in sets) {
+        final Iterable<String> filePaths = set.map((Command e) => e.filePath);
+        if (!filePaths.contains(command.filePath)) {
```

Comment:
Would the set math be easier making the parameter literally a set `Iterable<Set<Command>>`?

Reply:
Like can we get rid of some of this code by taking advantage of Set?  I don't think so.  We'd have to add hashing ability to Command and make clear what it means to look up a Command by file path.  So in that sense it will make the code more complicated.  It might make things a bit faster however.

Reply:
Done, moved to `Set<String>` to speed up intersection calculation.